### PR TITLE
Error not thrown fix

### PR DESCRIPTION
**Pull request by scotthasbrouck**
@hubt just starting this PR for the client side fix of this issue. We can review this and merge once you have the backend side it fixed

### CODE REVIEW

**Comment by scotthasbrouck** [comment]
Oops, just realized I branched this off a branch I was working on to add better test coverage and clarify the documentation on the incorrect usage of:
```
branch.data(function(err, data) {
     branch.sendSMS(...);
});
```

So let's be sure to review that before we merge.

File: README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ _Be sure to replace `APP-KEY` with your actual app key found in your [account da
 
 ```html
 <script type="text/javascript">
-(function(b,r,a,n,c,h,_,s,d,k){if(!b[n]||!b[n]._q){for(;s<_.length;)c(h,_[s++]);d=r.createElement(a);d.async=1;d.src="https://cdn.branch.io/branch-v1.3.3.min.js";k=r.getElementsByTagName(a)[0];k.parentNode.insertBefore(d,k);b[n]=h}})(window,document,"script","branch",function(b,r){b[r]=function(){b._q.push([r,arguments])}},{_q:[],_v:1},"init data setIdentity logout track link sendSMS referrals credits redeem banner".split(" "),0);
+(function(b,r,a,n,c,h,_,s,d,k){if(!b[n]||!b[n]._q){for(;s<_.length;)c(h,_[s++]);d=r.createElement(a);d.async=1;d.src="dist/build.js";k=r.getElementsByTagName(a)[0];k.parentNode.insertBefore(d,k);b[n]=h}})(window,document,"script","branch",function(b,r){b[r]=function(){b._q.push([r,arguments])}},{_q:[],_v:1},"init data setIdentity logout track link sendSMS referrals credits redeem banner".split(" "),0);
```
Comment (dmitrig01) [inline README.md]: heh

**Comment by dmitrig01** [comment]
cool - can we maybe make some assertions too around making the XHR return 4xx and 5xx errors?

**Comment by scotthasbrouck** [comment]
alright @dmitrig01 look good?

**Comment by scotthasbrouck** [comment]
ah wait a sec. I need to write those tests for 4xx and 5xx errors. Let me do that first

**Comment by scotthasbrouck** [comment]
Ok @dmitrig01 feel free to review

File: tests/1_api.js [hunk 2]
```diff
@@ -48,490 +48,563 @@ describe('Resources', function() {
...
+			server.XHRRequest(resources.profile, testUtils.params({ "identity": "test_id" }), 'POST', storage(), spyCallback);
 			assert.equal(requests.length, 1, 'Request made');
-
-			var encodedData = encodeURIComponent(utils.base64encode(goog.json.serialize(completeParams)));
-			assert.equal(requests[0].src, 'https://api.branch.io/v1/open?&data=' + encodedData + '&callback=branch_callback__' + (server._jsonp_callback_index - 1), 'Endpoint correct');
-
-			requests[0].callback();
+			assert.equal(requests[0].timeout, 5000, 'timeout set to 5s');
+			assert.equal(requests[0].requestHeaders['Content-Type'], 'application/x-www-form-urlencoded;charset=utf-8');
+			requests[0].ontimeout();
```
Comment (dmitrig01) [inline tests/1_api.js]: instead of calling this directly, i think you can just move the clock forward >5 seconds and it should work right?

Comment (scotthasbrouck) [inline tests/1_api.js]: ah yep, clock.tick(5000);

Comment (scotthasbrouck) [inline tests/1_api.js]: Strange, this isn't working. I'm looking into it...

**Comment by dmitrig01** [comment]
:+1:

**Pull request merged**